src/listwidget.cpp

```diff
@@ -26,6 +26,10 @@ void ListWidget::stfl_replace_lines(const ListFormatter& listfmt)
 {
 	num_lines = listfmt.get_lines_count();
 	form.modify(list_name, "replace_inner", listfmt.format_list());
+
+	if (num_lines > 0 && get_position() >= num_lines) {
+		set_position(num_lines - 1);
+	}
```

Comment:
It bothers me a bit that nothing will happen if `num_lines` is zero. This won't trigger the bug (because `PbView` has an `i >= 1` check), but still, I'd change this to:

```suggestion
	if (get_position() >= num_lines) {
		set_position(std::max(0, num_lines - 1));
	}
```

I also think that `stfl_replace_list()` above needs a similar block, to make the two methods behave similarly. And we should probably have a test for this new behaviour ;)

Reply:
Good points, done

Reply:
`std::max(0, num_lines - 1)` doesn't work here because `num_lines` is an unsigned integer so `num_lines - 1` will wrap around when `num_lines == 0`.